src/store/models/modals.ts

```diff
@@ -235,6 +253,102 @@ const modalsModel: ModalsModel = {
       defaultCommand: undefined,
     });
   }),
+  setBalanceChannels: action((state, payload) => {
+    state.balanceChannels = {
+      ...state.balanceChannels,
+      ...payload,
+    };
+  }),
+  showBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: true });
+  }),
+  hideBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: false });
+  }),
+  resetBalanceChannels: thunk(
+    async (actions, network, { getStoreActions, getStoreState }) => {
+      const channels = [] as LightningNodeChannel[];
+      const { getChannels } = getStoreActions().lightning;
+      const { links } = getStoreState().designer.activeChart;
+
+      const id2Node = {} as Record<string, LightningNode>;
+      const promisesToAwait = [] as Promise<unknown>[];
+      const channelsInfo = [] as ChannelInfo[];
+
+      for (const node of network.nodes.lightning) {
+        promisesToAwait.push(
+          getChannels(node).then((nodeChannels: LightningNodeChannel[]) => {
```

Comment:
I mentioned in the prior review to avoid `.then` and use `async/await` instead to be consistent with the rest of the codebase. I don't see a strong reason why `.then` is necessary in any of the places you've use it in this PR.

Reply:
You do not need to push promises into an array. The `for`/`of` loop will wait for the promises to resolve before continuing execution. This would work fine:
```ts
      for (const node of network.nodes.lightning) {
        const nodeChannels = await getChannels(node);
        channels.push(...nodeChannels);
        id2Node[node.name] = node;
      }
```